pkg/binding/bindingpolicy.go

```diff
@@ -88,6 +88,47 @@ func (c *Controller) syncBindingPolicy(ctx context.Context, bindingPolicyName st
 		logger.V(5).Info("Enqueued Binding for syncing, while handling BindingPolicy", "name", bindingPolicy.Name)
 		c.enqueueBinding(bindingPolicy.GetName())
 
+		statusCollectorName, err := c.identifyMissingStatusCollector(bindingPolicy)
+		if err != nil {
+			return fmt.Errorf("failed to list StatusCollector %s for BindingPolicy %s: %w", statusCollectorName, bindingPolicyName, err)
+		}
+
+		// If at least one StatusCollector object is missing, ensure the Condition and requeue the BindingPolicy,
```

Comment:
A missing StatusCollector should not cause requeuing. A missing StatusCollector should cause the proper error report in status, that is the successful reaction to the current state. Should a relevant StatusCollector object be created later, just as if one is deleted later, this is cause for reconsidering the current state.

Reply:
Although GitHub marks this as "Outdated", it is still relevant.

Reply:
Removed the requeuing.